pysat/_instrument.py

```diff
@@ -1003,8 +1003,9 @@ def __setitem__(self, key, new_data):
                     # (including list or slice).
                     self.data.loc[self.data.index[key[0]], key[1]] = new
 
-                self.meta._data_types[key[1]] = self.data[
-                    key[1]].values.dtype.type
+                if not np.isnan(new):
+                    self.meta._data_types[key[1]] = self.data[
+                        key[1]].values.dtype.type
```

Comment:
```suggestion
                if np.isnan(new):
                    self.meta._data_types[key[1]] = float
                else:
                    self.meta._data_types[key[1]] = self.data[
                        key[1]].values.dtype.type
```

Is this a better way to move forward?

Reply:
I still get the same error when new is a nan:
```
TypeError: 'NoneType' object does not support item assignment
```

Reply:
Actually, if I change the set function to 37.0 rather than nan, I still get the same error.

Reply:
Found the source of the bug deeper in.  Updates pending the local tests...